src/main/java/baseball/GameManager.java

```diff
@@ -0,0 +1,71 @@
+package baseball;
+
+import baseball.generator.RandomNumberGenerator;
+import baseball.receiver.ConsoleInputReceiver;
+import baseball.receiver.InputReceiver;
+import baseball.number.GameNumber;
+import baseball.printer.ConsolePrinter;
+import baseball.printer.Printer;
+import baseball.result.Result;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class GameManager {
+    private final Printer printer;
+    private final InputReceiver receiver;
+    private final GameNumber seedNumber;
+    private final GameNumber userNumber;
+
+    public GameManager(Scanner scanner) {
+        if(scanner == null)
+            throw new NullPointerException();
+
+        printer = new ConsolePrinter();
+        receiver = new ConsoleInputReceiver(scanner);
+        seedNumber = new GameNumber();
+        userNumber = new GameNumber();
+    }
+
+    public void run() {
+        Result result = new Result();
+        boolean continueGame = true;
+
+        printer.greet();
+        printer.askStartGame();
+        continueGame = receiver.checkContinueGame();
+        while(continueGame) {
+            RandomNumberGenerator.generate(seedNumber);
+            startGame(result);
+
+            printer.noticeWin();
+            printer.askStartGame();
+            continueGame = receiver.checkContinueGame();
+        }
+    }
+
+    private void startGame(Result result) {
+
+        while(!result.isThreeStrike()) {
```

Comment:
컨벤션! 
자동 줄 정렬 기능 (Reformat code) 를 활용하면 IDE가 한방에 정렬 해줍니다 🙂

Reply:
8.1. 공백으로 줄을 끝내지 않음
이 컨벤션에 해당하는 건가요?
그러면 이 컨벤션에서는 가독성을 위한 공백은 허용하지 않는것인가요?
아니면 코드 블럭의 의미 구분을 위한 공백은 허용하나요?

Reply:
```suggestion
        while (!result.isThreeStrike()) {
```
을 의미 했었습니다!

8.1의 no-trailing-spaces 는 빈 줄에 공백을 넣는 행위를 의미하는것으로 보여지네요!
```
 
```
```
	
```
(보일진 모르겠지만 1 space와 1 tab이 들어있습니다.)

가독성을 위한 공백, 의미 구분을 위한 공백 전부 허용됩니다.
불필요한 공백만 아니면 됩니다 🙂

클린코드에 나오는 내용인데 관련하여 정리된 블로그글이 있어 공유드립니다.

[Clean Code -형식 맞추기-](https://medium.com/@ish453525/clean-code-%ED%98%95%EC%8B%9D-%EB%A7%9E%EC%B6%94%EA%B8%B0-b6ee86a45e99)

블로그글엔 소개되지 않았지만, 책에 나와있는 _내려가기 규칙_ 도 개인적으로 감명 깊게 봤던 부분이니 책이 도착하면 읽어보시기 바랍니다 🙂